File: src/java/TakeHome/src/test/java/docugami/takehome/tests/QueueTests.java
// Copyright (c) Docugami, Inc. All rights reserved.
package docugami.takehome.tests;

import org.junit.Assert;
import org.junit.Test;
import docugami.takehome.implementation.MyQueue;
public class QueueTests {

    public QueueTests() {
        super();
    }

    @Test
    public void testPeek() {
        MyQueue<Integer> queue = new MyQueue<Integer>(Integer.class, 5);
        queue.enqueue(5);
        Assert.assertEquals((Integer)5, queue.peek());
    }
}

File: src/csharp/Implementation.Tests/QueueTests.cs
// Copyright (c) Docugami, Inc. All rights reserved.

namespace TakeHome.Implementation.Tests
{
    using TakeHome.Implementation;
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    public class QueueTests
    {

        [TestMethod]
        public void TestPeek()
        {
            var queue = new Queue<int>(10);

            queue.Enqueue(1);
            Assert.IsTrue(queue.Peek() == 1);
        }
    }
}
